src/Compiler.php

```diff
@@ -5336,13 +5336,124 @@ public function findImport($url)
             }
         }
 
-        if ($urls) {
-            if (! $hasExtension || preg_match('/[.]scss$/', $url)) {
-                throw $this->error("`$url` file not found for @import");
-            }
+        throw $this->error("`$url` file not found for @import");
+    }
+
+    private function resolveImportPath($url, $baseDir)
```

Comment:
This method (and the following helper methods) are implementing the logic at https://github.com/sass/dart-sass/blob/28a582f45e3f9f3384071a83db2012b4e2e16707/lib/src/importer/utils.dart#L41-L44 with 2 exceptions (not counting the fact that dart-sass makes the `path.join()` in the caller):

- support for the `.sass` extension, as we don't support parsing them (this means we don't consider them as a source of conflict either)
- support for `.import.scss` files, as that's part of implementing Sass modules (this one will become support when/if we implement modules, but the code will probably have changed quite a bit at that point anyway)